src/Microsoft.Extensions.Caching.Memory/MemoryCache.cs

```diff
@@ -113,41 +120,34 @@ private void SetEntry(CacheEntry entry)
             // Initialize the last access timestamp at the time the entry is added
             entry.LastAccessed = utcNow;
 
-            var added = false;
             CacheEntry priorEntry;
-
-            _entryLock.EnterWriteLock();
-            try
+            if (_entries.TryRemove(entry.Key, out priorEntry))
```

Comment:
From what I understand it can update null, but it cannot update when there is no entry at all. (https://github.com/dotnet/corefx/blob/master/src/System.Collections.Concurrent/src/System/Collections/Concurrent/ConcurrentDictionary.cs#L577)